### PR TITLE
Simple 2-layer neural network from scratch

### PR DESCRIPTION
**Pull request by venkywonka**
This PR implements a simple neural network with the following features:
* Trains for learning the XOR function
* has a 2-input, 2-hidden and 1-output node architecture
* uses backpropagation algorithm using momentum-based gradient descent
* own matrix implementation from scratch (no numpy!)
* 100 epochs

### CODE REVIEW

**Comment by Graey** [comment]
wow, thank you for this addition!
